authentication/auth_method_oidc.go

```diff
@@ -65,19 +68,43 @@ func (a oidcAuth) getMSALToken(ctx context.Context, api environments.Api, _ auto
 		return &authWrapper.Authorizer{Authorizer: conf.TokenSource(ctx)}, nil
 	}
 
+	idToken := a.idToken
+
+	if a.idTokenFilePath != "" {
+		idToken, err = a.readTokenFile(a.idTokenFilePath)
+
+		if err != nil {
+			return nil, fmt.Errorf("reading token file: %v", err)
+		}
+
+		if a.idToken != "" && a.idToken != idToken {
+			return nil, fmt.Errorf("mismatch between supplied OIDC token and supplied OIDC token file")
```

Comment:
ie
```suggestion
			return nil, fmt.Errorf("mismatch between supplied OIDC token (%s) and supplied OIDC token file contents (%s) - please either remove one or ensure they match", a.idToken, idToken)
```

Reply:
I'm not sure that we should to be honest - we go to pains to exclude tokens from log output in other places.